frontend/packages/dev-console/src/components/topology/nodeContextMenu.tsx

```diff
@@ -36,21 +36,26 @@ const createMenuItems = (actions: KebabMenuOption[]) =>
       />
     ),
   );
+};
 
 export const workloadContextMenu = (element: Node) =>
   createMenuItems(kebabOptionsToMenu(workloadActions(element.getData())));
 
-export const groupContextMenu = (element: Node) => {
+export const groupContextMenu = (element: Node, connectorSource?: Node) => {
   const applicationData: TopologyApplicationObject = {
     id: element.getId(),
     name: element.getLabel(),
     resources: element.getChildren().map((node: GraphElement) => node.getData()),
   };
 
-  return createMenuItems(kebabOptionsToMenu(groupActions(applicationData)));
+  return connectorSource
+    ? createMenuItems(groupActions(applicationData, connectorSource))
```

Comment:
This needs a `TODO` comment for cleanup or a fix now.
We shouldn't have to avoid calling `kebabOptionsToMenu`. Instead the actions provided should omit including the `path` when used in this scenario.

Reply:
fixed